PallyPower/libs/LibTalentProcess-1.0/LibTalentProcess-1.0.lua

```diff
@@ -57,10 +57,27 @@ function lib:ProcessTalents(talents)
         if(talents[1] > talents[2] and talents[1] > talents[3]) then
             talents.role = "DRUID_CASTER"
         elseif(talents[2] > talents[1] and talents[2] > talents[3]) then
-            -- TODO: Druid Off-/Tank/Melee Unterscheidung?
             talents.role = "DRUID_MELEE"
-            talents.role = "DRUID_OFF_TANK"
-            talents.role = "DRUID_TANK"
+
+            -- TODO: Druid Off-/Tank/Melee Unterscheidung?
+            local health = UnitHealthMax(talents.unitID)
+            local base, effectiveArmor, armor, posBuff, negBuff = UnitArmor(talents.unitID);
+            local baseDefense, armorDefense = UnitDefense(talents.unitID);
+            
+            if health >= 10000 then
+                if effectiveArmor >= 20000 then
+                    if (baseDefense + armorDefense) >= 380 then
+                        talents.role = "DRUID_TANK"
+                    else
+                        talents.role = "DRUID_OFF_TANK"
+                    end
+                else
+                    talents.role = "DRUID_OFF_TANK"
+                end
+            else
+                talents.role = "DRUID_OFF_TANK"
+            end
+
```

Comment:
Der check funktioniert nur dann sinnvoll, wenn der Character direkt in Bärform ist während er joint oder täusche ich mich da? Ohne Bärenform kommt der Spieler sicher nicht auf 20000 Rüstung. Heißt hier müsste noch geprüft werden ob der Spieler Druide ist und POWER = RAGE, wenn nein dann den Faktor nehmen (3.8 oder was das war?) auf die HP und Rüstung.
Dann sollte der Check sinnvoller funktionieren.

Reply:
Ich hab mal ohne die Faktoren gearbeitet, da diese unterschiedlich ausfallen können.
Laut Tooltip +25% Stamina und +400% Armor, rechnerisch kam 1,5 bzw. 4,48 raus.